src/encode.cpp

```diff
@@ -412,8 +410,9 @@ optional<string> encodeOp(State &st, mlir::memref::BufferCastOp op) {
 
   } else {
     vector<Expr> idxs = createBoundIndexVars(memrefTy.getRank());
-    auto tVal = tensor.get(idxs);
-    auto [mVal, success] = memref.load(idxs);
+    auto [tVal, tSuccess] = tensor.get(idxs);
+    auto [mVal, mSuccess] = memref.get(idxs);
+    auto success = tSuccess & mSuccess;
```

Comment:
Is there a case where tSuccess != mSuccess?

Reply:
I think `mSuccess` is more strict than `tSuccess`.
```
mSuccess = dimension inbounds && bid, offset inbounds
tSuccess = dimension inbounds
```
Here buffer_cast op only accept when tensor, memref dimensions are match.
So the first term will be same in both success.